File: Dynamic-program/62-uniquePaths/62.py
class Solution:
    def uniquePaths(self, m: int, n: int) -> int:
        board = [[0] * n for _ in range(m)]
        for i in range(m):
            for j in range(n):
                if i == 0 and j == 0:
                    board[i][j] = 1
                elif i == 0:
                    board[i][j] = board[i][j - 1]
                elif j == 0:
                    board[i][j] = board[i - 1][j]
                else:
                    board[i][j] = board[i - 1][j] + board[i][j - 1]
        return board[-1][-1]

File: Dynamic-program/62-uniquePaths/go/62.go
package main

import "fmt"

func uniquePaths(m int, n int) int {
	dp := make([][]int, m)
	for i := range m {
		dp[i] = make([]int, n)
		dp[i][0] = 1
	}
	for j := range n {
		dp[0][j] = 1
	}

	for i := 1; i < m; i++ {
		for j := 1; j < n; j++ {
			dp[i][j] = dp[i-1][j] + dp[i][j-1]
		}
	}

	return dp[m-1][n-1]
}

func main() {
	fmt.Println(uniquePaths(3, 7))	
}
